src/components/Tool/FixCommitmentInfo/Contact.tsx

```diff
@@ -79,225 +125,472 @@ const useStyles = makeStyles()(() => ({
-            </Box>
-            <Box>
-              <IconButton
-                data-testid="goToContactsButton"
-                onClick={() =>
-                  push({
-                    pathname: `/accountLists/[accountListId]/contacts/[contactId]`,
-                    query: { accountListId, contactId: id },
-                  })
-                }
-              >
-                <SearchIcon />
-              </IconButton>
-              <IconButton
-                data-testid="hideButton"
-                onClick={() => hideFunction(id)}
-              >
-                <VisibilityOffIcon />
-              </IconButton>
-            </Box>
-          </Box>
-        </Grid>
-      </Grid>
+      <Formik
+        initialValues={{
+          statusValue: statusValue,
+          pledgeCurrency: amountCurrency,
+          pledgeAmount: amount || null,
+          pledgeFrequency: frequencyValue,
+        }}
+        validationSchema={appealFormSchema}
+        onSubmit={async (values) => {
+          await onSubmit(values);
+        }}
+      >
+        {({
+          values: {
+            statusValue,
+            pledgeCurrency,
+            pledgeAmount,
+            pledgeFrequency,
+          },
+          handleSubmit,
+          setFieldValue,
+          errors,
+        }): ReactElement => {
+          const modalContact = {
+            id: id,
+            status: statusValue,
+            name: name,
+            pledgeCurrency,
+            pledgeAmount,
+            pledgeFrequency,
+          };
+          return (
+            <Form onSubmit={handleSubmit}>
+              <Grid container className={classes.formWrapper}>
+                <Card className={classes.formInner}>
+                  <Grid container>
+                    <Grid item lg={4} md={4} xs={12}>
+                      <Box
+                        display="flex"
+                        p={2}
+                        alignItems="center"
+                        className={classes.left}
+                      >
+                        <ContactAvatar
+                          src={avatar || ''}
+                          aria-label="Contact Avatar"
+                        />
+                        <Box display="flex" flexDirection="column" ml={2}>
+                          <Link
+                            data-testid="contactSelect"
+                            underline="hover"
+                            onClick={() => setContactFocus(id, 'Donations')}
+                          >
+                            <Typography variant="subtitle1">{name}</Typography>
+                          </Link>
+                          <Typography variant="subtitle2">
+                            {`Current: ${statusTitle} ${
+                              amount && amountCurrency
+                                ? currencyFormat(amount, amountCurrency, locale)
+                                : ''
+                            } ${frequencyTitle}`}
```

Comment:
Check that frequency is defined before displaying it. Sometimes it prints 'undefined'
<img width="318" alt="Screenshot 2024-08-09 at 8 21 04 AM" src="https://github.com/user-attachments/assets/c3165e84-3598-4650-9d0b-8965c67df56a">

Reply:
@wjames111 this is still showing as undefined for me. I think it has to do with the list of `frequencies`. Let's use this way of getting the frequency list. This code is from line 482 of `EditPartnershipInfoModal.tsx`
```
                    {Object.values(PledgeFrequencyEnum).map((value) => (
                      <MenuItem key={value} value={value}>
                        {getLocalizedPledgeFrequency(t, value)}
                      </MenuItem>
                    ))}
```

Reply:
@wjames111 try setting a contacts frequency to `annual` and it will probably show `undefined`.